### ISSUE TITLE
Word Frequency fails to position cursor

### ISSUE THREAD

**Issue by okrick** [DistributedProofreaders/guiguts-py#628]
![Capture](https://github.com/user-attachments/assets/b2102dec-6a4a-4110-9cb2-a9ab20087f18)

Ran the Alpha/Num check.  Except for `000.jpg`, selecting any item on the list automatically positions the cursor at the beginning of the item in the document.

[travels.zip](https://github.com/user-attachments/files/18283866/travels.zip)

**Comment by srjfoo** [comment]
@okrick, What was the expected result?

**Comment by okrick** [comment]
I expected that `000.jpg` would be selected, highlighted, and the item positioned in the editing window.  Instead, I got a search failed warning beep.

**Comment by srjfoo** [comment]
> I expected that `000.jpg` would be selected, highlighted, and the item positioned in the editing window. Instead, I got a search failed warning beep.

Ah, I see. I had to turn off "whole word only" in the search form, but when I cmd+left-clicked on it in WF, the search window came up and find all found it. I wonder if WF is confused by the prefix; I didn't see anything in the log.

**Comment by okrick** [comment]
Yet, 001.png etc. are selectable.  The only one on the list not selectable is 000.jpg.

**Comment by srjfoo** [comment]
> Yet, 001.png etc. are selectable. The only one on the list not selectable is 000.jpg.

000.jpg is also the only one that has a prefix that would cause the assumption that it's not a "full word". The "whole word" checkbox appears to be checked by default when a search form is brought up using the cmd/ctrl+left click, presumably to omit partial matches). 

I can cmd-left-click on `000.jpg`, and the search form is brought up with that filled in. If I leave "whole word" checked, there are no results. If I un-check "whole word", I get two results.

Now, the question is -- should WF be grabbing the whole illustration name, i.e. `i_000.jpg`? I think if it had, you wouldn't be seeing this issue.

**Comment by okrick** [comment]
Yes.  Either the `i_` should be included in the WF or the Search & Replace tool should not consider the `i_` as part of the word.  They should have the same definition.

**Comment by windymilla** [comment]
All the above comments from @okrick & @srjfoo are completely fair: there was a problem with word boundaries and underscores. 

Because the origins of regexes are computing-based, the characters that count as part of a word include the underscore character. So, when you do a "whole word" search, and it tries to find 
"<non-word char><some word chars><non-word char>",
if the character before the "real" word, like "dog" is an underscore, e.g. "_dog!", it won't be found because the character before "dog" is an underscore, so (in regex's mind) it's a word character.

However, certainly as far as WF is concerned, we don't want a leading or trailing underscore to be included in a "word", mostly because we use them to mark italics, e.g. "_dog_". So the definition of a "word" for WF is essentially words separated by space, with things like punctuation removed.

I have implemented a solution which tackles this issue - see #634 for details.
